src/distributions/mod.rs

```diff
@@ -174,13 +174,19 @@ fn ziggurat<R: Rng+?Sized, P, Z>(
         // this may be slower than it would be otherwise.)
         // FIXME: investigate/optimise for the above.
         let bits: u64 = uniform(rng);
-        let i = (bits & 0xff) as usize;
-        let f = (bits >> 11) as f64 / SCALE;
 
         // u is either U(-1, 1) or U(0, 1) depending on if this is a
         // symmetric distribution or not.
-        let u = if symmetric {2.0 * f - 1.0} else {f};
-        let x = u * x_tab[i];
+        // FIXME: the distribution is not open, but half-open.
+        //        Can that cause problems or a bias?
```

Comment:
I have no idea, sorry. But would rejection sampling slow it down much anyway, in practice? Maybe it's not significant enough to bother anyway.

Reply:
I have been reading up on the ziggurat method. There even is an improved version from 2015, that should be faster. But I added the note to make sure it didn't get lost